scipy/spatial/_spherical_voronoi.py

```diff
@@ -185,6 +228,14 @@ def _calc_vertices_regions(self):
         Tyler Reddy, Ross Hemsley and Nikolai Nowaczyk
         """
 
+        # test degenerate input
+        rank = np.linalg.matrix_rank(self.points - self.center)
```

Comment:
we may also want to use the `tol` argument if we do this--I assume there's some kind of precision limit on determining whether all points lie exactly on a great spherical arc, and `SphericalVoronoi` currently uses `threshold` argument to give a user-tunable control on other related checks

This also reminds me of a potentially useful unit test -- the case of geodesic input where just one of the points is ever-so-slightly nudged off the great circle arc--this should be handled just fine without the interception code handled here, right? Indeed, if the resulting Voronoi diagram from that input looks very similar to the approach proposed here that may increase our confidence.

Reply:
This shows the Voronoi regions of points on the plane, as handled in this PR:
![unperturbed](https://user-images.githubusercontent.com/5646170/64464261-4f62d780-d0d5-11e9-99a5-ed431c02506c.png)

Here I have perturbed all the input points, such that the code added in this PR is not invoked.  The shared vertices at the north and south poles are split up, but it is otherwise the same solution:
![perturbed](https://user-images.githubusercontent.com/5646170/64464266-52f65e80-d0d5-11e9-8b66-dab91d4dee2d.png)

Reply:
I have used the tolerance parameter as you suggested.  I also updated the `max_discrepancy` check so that it uses Euclidean distances, rather than squared distances.  Now all checks (discrepancy, duplicates, rank) are based on Euclidean distances.

Reply:
This looks promising! I'm interested in a unit test for the "slightly-perturbed" case as well, though I suppose that isn't strictly required here. Just helps me see that near-geodesic input is handled well & similar to the new geodesic code path here as images above show.

I should maybe run the benchmarks on this feature branch vs. master---hopefully the extra `matrix_rank` check isn't too expensive for large inputs.